aws_lambda_powertools/utilities/data_classes/kinesis_firehose_event.py

```diff
@@ -1,9 +1,125 @@
 import base64
-from typing import Iterator, Optional
+import json
+from dataclasses import dataclass, field
+from typing import Any, Callable, Dict, Iterator, List, Optional
+
+from typing_extensions import Literal
 
 from aws_lambda_powertools.utilities.data_classes.common import DictWrapper
 
 
+@dataclass(repr=False, order=False, frozen=True)
+class KinesisFirehoseDataTransformationRecordMetadata:
+    """
+    Documentation:
+    --------------
+    - https://docs.aws.amazon.com/firehose/latest/dev/dynamic-partitioning.html
+    """
+
+    partition_keys: Optional[Dict[str, str]]
```

Comment:
```suggestion
    partition_keys: Optional[Dict[str, Any]]
```

to match your example with "year": 2023

Reply:
Sorry this is a mistake by me, We shouldn't use int in the Map. Although it's not clearly documented in the Firehose's doc, but every example they provide is in `Dict[str, str]` format. So reverting this back to str:str and changed 2023->"2023" in test case